### PR TITLE
Loading Screen added

### LINKED ISSUE #101: Loading Widget
### Is your feature request related to a problem? Please describe.

Everywhere we have add circularloading icon now we need to replace it with our ecell animation widget.

### Describe the solution you'd like

Make a loading widget and add ot everywhere in buildLodaing function.

### PR DESCRIPTION
**Pull request by Hard-Coder05**
### Description

**Type of Change:** (Delete irrelevant options)
- Code 
- User Interface 

**Code/Quality Assurance Only** (Delete irrelevant options)
 
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?

tested locally

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged
- [ ] I have added one/more required dependencies
- [x] I have formatted the code using dart format (dartfmt)

**Code/Quality Assurance**
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by werainkhatri — CHANGES_REQUESTED** [review]
1. remove loading_screen.dart
2. move the background widget from `buildSuccess` to `build`
3. add the ecellanimation to buildLoading
4. test it in all the screen once
5. revert pubspec.lock changes

**Comment by Hard-Coder05** [comment]
@werainkhatri Done from my side

File: lib/screens/events/events.dart [hunk 1]
```diff
@@ -28,29 +27,55 @@ class EventsScreen extends StatelessWidget {
...
 
-  Widget _buildSuccess(BuildContext context, List<Event> eventsList) {
-    //TODO On success UI
+  Widget _buildLoading(BuildContext context) {
+    double width = MediaQuery.of(context).size.width;
+    return Stack(
+      children: [
+        Center(child: ECellLogoAnimation(size: width / 2)),
+      ],
+    );
```
Comment (werainkhatri) [inline lib/screens/events/events.dart]: Single widget in stack can be replaced with just the element widget. Do the changes in other screens too.
```suggestion
    return Center(child: ECellLogoAnimation(size: width / 2));
```

File: lib/screens/forgot_password/forgot_password.dart [hunk 2]
```diff
@@ -25,21 +25,24 @@ class ForgotPasswordScreen extends StatelessWidget {
...
+              else if (state is ForgotPasswordError)
+                _uiUpdateForNetworkError(context, state.state)
+              else if (state is ForgotResetInitial)
+                _resetPassword(context, state)
+              else if (state is ForgotResetSuccess)
+                _passwordResetSuccess()
+              else
+                _initialForgotPassword(context, state)
+            ],
+          );
```
Comment (werainkhatri) [inline lib/screens/forgot_password/forgot_password.dart:45]: whats the difference b/w this and the previous code?

File: lib/screens/sponsors/sponsors.dart [hunk 3]
```diff
@@ -18,24 +18,19 @@ class SponsorsScreen extends StatelessWidget {
...
-            return _buildLoading();
-          } else {
-            return _buildAskReload();
-          }
+          return Stack(
+            children: [
+              if (state is SponsorsInitial) _buildLoading(context),
+              if (state is SponsorsSuccess) _buildSuccess(context, state.sponsorsList),
+              if (state is SponsorsLoading) _buildLoading(context),
+              if (state is SponsorsError) _buildAskReload(),
```
Comment (werainkhatri) [inline lib/screens/sponsors/sponsors.dart]: if...else it

**Review by werainkhatri — CHANGES_REQUESTED** [review]

File: lib/screens/speaker/speaker.dart [hunk 4]
```diff
@@ -26,27 +26,39 @@ class SpeakerScreen extends StatelessWidget {
...
+                  child: ((() {
+                    if (state is SpeakerInitial)
+                      return _buildLoading(context);
+                    else if (state is SpeakerSuccess)
+                      return _buildSuccess(context, state.speakerList);
+                    else if (state is SpeakerLoading)
+                      return _buildLoading(context);
+                    else
+                      return _buildAskReload();
+                  }())),
```
Comment (werainkhatri) [inline lib/screens/speaker/speaker.dart]:
```suggestion
                  child: () {
                    if (state is SpeakerInitial)
                      return _buildLoading(context);
                    else if (state is SpeakerSuccess)
                      return _buildSuccess(context, state.speakerList);
                    else if (state is SpeakerLoading)
                      return _buildLoading(context);
                    else
                      return _buildAskReload();
                  }(),
```

**Review by werainkhatri — APPROVED** [review]

**Pull request merged**